addons/base_onboarding/models/base_onboarding_state.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+ONBOARDING_STEP_STATES = [
+    ('not_done', 'Not done'),
+    ('just_done', 'Just done'),
+    ('done', 'Done'),
+]
+ONBOARDING_STATES_ADDITIONAL_STEPS = [('closed', "Closed")]
+ONBOARDING_STATES = ONBOARDING_STEP_STATES + ONBOARDING_STATES_ADDITIONAL_STEPS
+
+
+class OnboardingStateMixin(models.AbstractModel):
+    """Fields and methods common to onboarding and onboarding step states """
+    _name = 'base.onboarding.state.mixin'
```

Comment:
Have to say this is not really clear ... proposing instead a table to store the actual usage of onboarding, like ``onboarding.progress``, with onboarding_id on it, company_id, user-id (not now, but if some day we make a per-user onboarding), ...